tests/utils.go

```diff
@@ -273,7 +281,7 @@ func GetConnectionWithOptions(options *clickhouse.Options) (driver.Conn, error)
 	return clickhouse.Open(options)
 }
 
-func getConnection(env ClickHouseTestEnvironment, database string, settings clickhouse.Settings, tlsConfig *tls.Config, compression *clickhouse.Compression) (driver.Conn, error) {
+func getConnection(env ClickHouseTestEnvironment, database string, roUser bool, settings clickhouse.Settings, tlsConfig *tls.Config, compression *clickhouse.Compression) (driver.Conn, error) {
```

Comment:
Nit: we need something more powerful than `bool`. There are 3 modes for `readonly` users https://clickhouse.com/docs/en/operations/settings/permissions-for-queries/#readonly

Reply:
@mshustov definitely. However, I would start with having a read-only user covered. Please take a look at the recent changes.